app/Http/Controllers/Admin/RoleController.php

```diff
@@ -0,0 +1,336 @@
+
+                DB::commit();
+
+                return response()->json([
+                    'error' => false,
+                    'message' => 'Success !',
+                ]);
+            }
+        } catch (Exception $e) {
+            return response()->json([
+                'error' => true,
+                'message' => 'Fail !',
+            ]);
+        }
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        DB::beginTransaction();
+
+        try {
+            DB::table('roles')->where('id', $id)->delete();
+
+            DB::commit();
+
+            return response()->json([
+                'error' => false,
+                'message' => 'Success !',
+            ]);
+        } catch (Exception $e) {
+            DB::rollback();
+
+            return response()->json([
+                'error' => true,
+                'message' => 'Fail !',
+            ]);
+        }
+    }
+
+    /**
+     * [getRoles : Lấy ra danh sách vai trò]
+     * @return [type] [description]
+     */
+    public function getRoles()
+    {
+        $roles = $this->roleRepository->orderBy('id', 'desc')->get();
+
+        return Datatables::of($roles)
+            ->addIndexColumn()
+
+            ->editColumn('created_at', function ($role){
+                $time = $role->created_at;
+
+                $time_numb = strtotime($time);
+
+                return date('H:i | d-m-Y', $time_numb);
+            })
+
+            ->addColumn('action', function ($role) {
+                $string = '';
+
+                $string = $string .' <a href="'. route('roles.show', $role->id) .'" class="m-portlet__nav-link btn m-btn m-btn--hover-brand m-btn--icon m-btn--icon-only m-btn--pill" data-tooltip="tooltip" title="'. __('Permission') .'"><i class="la la-shield" aria-hidden="true"></i></a>' . '&nbsp;&nbsp;';
+
+                $string = $string . '&nbsp;&nbsp;' . '<a href="javascript:;" data-id="'. $role->id .'" class="m-portlet__nav-link btn m-btn m-btn--hover-brand m-btn--icon m-btn--icon-only m-btn--pill edit_role" data-tooltip="tooltip" title="'. __('Edit') .'"><i class="la la-edit" aria-hidden="true"></i></a>' . '&nbsp;&nbsp;';
+
+                $string = $string . '&nbsp;&nbsp;' . '<a href="javascript:;" data-id="'. $role->id .'" class="m-portlet__nav-link btn m-btn m-btn--hover-brand m-btn--icon m-btn--icon-only m-btn--pill delete_role" data-tooltip="tooltip" title="'. __('Delete') .'"><i class="la la-trash"></i></a>';
+
+                return $string;
+            })
+
+        ->make(true);
+    }
+
+    /**
+     * [getPermissionRole : Lấy những quyền hạn thuộc vai trò đó]
+     * 
+     * @param  [type] $role_id [description]
+     * @return [type]          [description]
+     */
+    public function getPermissionRole($role_id)
+    {
+        $role = $this->roleRepository->findOrFail($role_id);
+
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/RoleController.php</br>Variable "role_id" is not in valid camel caps format